ops/model.py

```diff
@@ -2043,6 +2522,18 @@ class RelationDataAccessError(RelationDataError):
     """
 
 
+class SecretPermissionError(ModelError):
+    """Raised when a secret operation is forbidden."""
+
+
+class SecretOwnershipError(SecretPermissionError):
+    """Raised when a secret holder attempts an owner-only operation, or vice versa."""
+
+
+class SecretNotGrantedError(SecretPermissionError):
```

Comment:
as discussed, the plan is to raise these instead of generic ModelErrors as soon as I know what specific error codes are returned by juju on failed hook tool calls
added a TODO to clarify